csrc/expr_simplifier.cpp

```diff
@@ -1490,10 +1478,8 @@ bool isPositiveHelper(Val* value, const Context& context) {
 
 bool isNonZero(Val* value, const Context& context) {
   value = foldConstants(value);
-  if (value->getInt().has_value() && *value->getInt() != 0) {
-    return true;
-  }
-  if (value->getDouble().has_value() && *value->getDouble() != 0.0) {
+  if (value->value().hasValue() && value->value().is<int64_t>() &&
+      value->value().as<int64_t>() != 0) {
```

Comment:
I think this should be
```suggestion
  if (value->value().hasValue() && value->value() != 0) {
```